src/scene/materials/material.js

```diff
@@ -68,6 +68,15 @@ class Material {
      */
     name = 'Untitled';
 
+    /**
+     * A unique id the user can assign to the material. The engine internally does not use this for
+     * anything, and the user can assign a value to this id for any purpose they like. Defaults to
+     * an empty string.
+     *
+     * @type {string}
+     */
+    userId = '';
```

Comment:
why not just use `material.name` for this?

Reply:
this is assigned to / managed by the engine. Ideally this can be used by a code and work even if the user renames the material for example.

Reply:
👍🏻 makes sense